packages/debridge/src/debridge.types.ts

```diff
@@ -0,0 +1,121 @@
+import { AsyncResult } from '@catalogfi/utils';
+import { JsonRpcSigner, Wallet } from 'ethers';
+import { DeBridgeTransaction, Estimation, Order } from './debridge.api.types';
+import { Transport, WalletClient } from 'viem';
+
+export interface IDebridge {
+  quote(
+    quoteConfig: QuoteConfig,
+    abortController: AbortController
+  ): AsyncResult<QuoteResult, DeBridgeErrorCodes>;
+  swap(swapConfig: SwapConfig): AsyncResult<SwapResponse, string>;
+  getTxs(
+    getTxsConfig: GetTxsConfig
+  ): AsyncResult<GetDebridgeTxsResponse, string>;
+  getTx(txHash: string): AsyncResult<GetDebridgeTxsResponse, string>;
+  getPoints(address: string): AsyncResult<DeBridgePoints, string>;
+}
+
+export type SwapTokenMetadata = {
+  chainId: number;
+  address: string;
+  decimals: number;
+};
+
+export type QuoteConfigWithoutAmount = {
+  fromToken: SwapTokenMetadata;
+  toToken: SwapTokenMetadata;
+  fromAddress: string;
+  toAddress: string;
+};
+
+export type QuoteConfigWithFromAmount = QuoteConfigWithoutAmount & {
+  fromAmount: string;
+};
+
+export type QuoteConfigWithToAmount = QuoteConfigWithoutAmount & {
```

Comment:
Instead of all these types, add amount and isExactOut fields in the QuoteConfigWithoutAmount.
QuoteConfigWithoutAmount = {
  fromToken: SwapTokenMetadata;
  toToken: SwapTokenMetadata;
  fromAddress: string;
  toAddress: string;
  amount:string;
  isExactOut:string;
}